tests/Resources/Page/OCPDashboard/OperatorHub/OperatorHub.robot

```diff
@@ -3,26 +3,31 @@ Library     SeleniumLibrary
 
 *** Keywords ***
 Install Operator
-    [Arguments]    ${operator}
+    [Arguments]    ${operator}      ${redhat_marketplace}=None
     Search Operator    ${operator}
-    Select Operator    ${operator}
+    Run Keyword If   "${redhat_marketplace}" == "None"    Select Non Marketplace Operator   ${operator}
+    ...         ELSE    Select Operator    ${operator}
     ${show_operator_warning_visible} =    Show Operator Warning Is Visible
     Run Keyword If    ${show_operator_warning_visible}   Confirm Show Operator
     Click Install
     Click Install
     Wait Until Installation Completes
 
-
 Search Operator
    [Arguments]    ${operator}
    Wait Until Element is Visible    //input[@data-test="search-operatorhub"]   timeout=150
    Input text    //input[@data-test="search-operatorhub"]   ${operator}
    Press keys    //input[@data-test="search-operatorhub"]   RETURN
 
+Select Non Marketplace Operator
```

Comment:
Probably, we would need to have a more generic keyword to install operator from whatever catalog (marketplace, certified operators, community, ibm etc)

Reply:
do you mean to create another keyword ?

Reply:
I mean having a keyword like "Select Operator" which takes as argument the operator name + the catalog name (marketplace or community for examples)

Reply:
hmm...i will modify the below keyword to handle the catlog value.I think this keyword should be there to handle non marketplace value

Reply:
isn't possible to handle non-marketplace using the same keyword? At the end, when you say non-marketplace you need an operator from a catalog different from mrktpl.

Reply:
you could do that ...however it will need nested If/else  block. It will be easy to understand and implement  if we have different keyword for both. Below is the code i am planning to add:

''''
Select Operator Without Respective Catalog Name Included
    [Arguments]    ${operator}      ${catalog}
    Wait Until Element is Visible    //a[contains(@data-test, "${operator}") and not (contains(@data-test," ${catalog}"))]   timeout=50
    Click Element    //a[contains(@data-test, "${operator}") and not (contains(@data-test," ${catalog}"))]

Select Operator with Catalog Name Included
    [Arguments]    ${operator}     ${catalog}
    Wait Until Element is Visible   //a[contains(@data-test, "${operator}") and (contains(@data-test," ${catalog}"))]  timeout=50
    Click Element    //a[contains(@data-test, "${operator}") and (contains(@data-test,"${catalog}"))]
'''

Reply:
maybe, `Select Operator with Catalog Name Included` is enough to do the job because you always know the catalog the operator must be fetched from

Reply:
Done